src/content/docs/Challenge/challenge-embedded.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Embedded Challenge
+description: Hardened JavaScript Escape Room on Embedded Hardware
+---
+
+<link rel="stylesheet" href="../index.css">
+
+The Embedded Challenge is a port of this site's [Web Challenge](https://hardenedjs.org/challenge/challenge) to [Moddable Six](https://www.moddable.com/moddable-six), an embedded device powered by the popular [ESP32-S3](https://www.espressif.com/en/products/socs/esp32-s3) microcontroller. The Embedded Challenge runs on the XS JavaScript engine. XS implements Hardened JavaScript natively, as part of the engine itself, including `lockdown()` and the `Compartment` class. The XS implementation of Hardened JavaScript is a completely separate implementation from the SES shim used by web-based Challenge.
+
+## Running the Embedded Challenge
+
+To run the Embedded Challenge, you'll need a Moddable Six (or equivalent) and a computer (macOS, Windows, or Linux) with the Moddable SDK and ESP-IDF installed. From there, it is straightforward to build, install, and run:
+
+```
+> cd $MODDABLE/contributed/challenge
+> mcconfig -d -m -p esp32/moddable_six_cdc
+```
+
+The user interface shows the secret code and latest guess. It also allows you to run the example attacker scripts from the Hardened JavaScript web page. Here's a video that shows the app running on Moddable Display 6:
+
+[<img src="https://img.youtube.com/vi/kUOTjM-oymw/hqdefault.jpg" width=75%/>](https://www.youtube.com/embed/kUOTjM-oymw)
+
+You can run your own attack scripts too.  To do that, you'll use a local web page and BLE to communicate with the Moddable Six.
+
+1. Start-up a web server
+
+	```
+	> cd $MODDABLE/contributed/challenge/web
+	> python3 -m http.server 8080
+	```
+
+2. Launch Chrome (Chrome is currently the only browser that implements Web BLE)
+3. Open the web page at `http://localhost:8080`
+4. Press the "Connect" button to establish a connection to the embedded device
+5. Enter your attack script
+6. Press the "Attack" button to run the script on the embedded device
+
+![mcsim](../../../assets/embedded-challenge-webpage.png)
+
+## Simulating the Embedded Challenge
+
+The Embedded Challenge also runs on macOS, Windows, and Linux using the `mcsim`, the Moddable SDK's simulator. This allows you to try out the Embedded Challenge while waiting for your embedded device to be delivered. To run the Embedded Challenge in the simulator, install the Moddable SDK and then:
+
+```
+> cd $MODDABLE/contributed/challenge
+> mcconfig -d -m -p sim/moddable_six
+```
+
+![mcsim](../../../assets/embedded-challenge-simulator.png)
+
+The application works exactly the same. However, BLE is unavailable so you cannot download your own attack scripts. You can edit the source code of built-in scripts, however, to try various attacks. While running in the simulator, the xsbug JavaScript debugger is available to debug your attacks and explore the runtime environment.
```

Comment:
XSBug could use a link or instructions on how to build and run it.

Reply:
Will add a link. It is built as part of setting up the Moddable SDK, so nothing special there.